### PR TITLE
add User service dao,AuthenticationService

### PR DESCRIPTION
**Pull request by AndrewLinkii**

### CODE REVIEW

File: src/main/java/cinema/dao/impl/UserDaoImpl.java [hunk 1]
```diff
@@ -0,0 +1,46 @@
...
+import org.hibernate.Session;
+import org.hibernate.Transaction;
+import org.hibernate.query.Query;
+
+@Dao
+public class UserDaoImpl implements UserDao {
+    @Override
+    public User add(User user) {
+        Session session = null;
+        Transaction transaction = null;
```
Comment (dmytro-hryhoruk) [inline src/main/java/cinema/dao/impl/UserDaoImpl.java:18]: move declaration to try block

File: src/main/java/cinema/dao/impl/UserDaoImpl.java [hunk 2]
```diff
@@ -0,0 +1,46 @@
...
+    }
+
+    @Override
+    public Optional<User> findByEmail(String email) {
+        try (Session session = HibernateUtil.getSessionFactory().openSession()) {
+            Query<User> query = session.createQuery("from User where email = :email", User.class);
+            query.setParameter("email", email);
+            return Optional.ofNullable(query.uniqueResult());
+        } catch (Exception e) {
+            throw new DataProcessingException("cant to find movie sessions by movie ", e);
```
Comment (dmytro-hryhoruk) [inline src/main/java/cinema/dao/impl/UserDaoImpl.java:43]: ?

Comment (dmytro-hryhoruk) [reply src/main/java/cinema/dao/impl/UserDaoImpl.java:43]: do not copy paste, that's pure evil

File: src/main/java/cinema/service/impl/AuthenticationServiceImpl.java [hunk 3]
```diff
@@ -0,0 +1,31 @@
...
+import cinema.exception.AuthenticationException;
+import cinema.lib.Inject;
+import cinema.model.User;
+import cinema.service.AuthenticationService;
+import cinema.service.UserService;
+import cinema.util.HashUtil;
+
+public class AuthenticationServiceImpl implements AuthenticationService {
+    @Inject
+    UserService userService;
```
Comment (dmytro-hryhoruk) [inline src/main/java/cinema/service/impl/AuthenticationServiceImpl.java:12]: access modifier

File: src/main/java/cinema/dao/impl/UserDaoImpl.java (hunk 2 above)
Comment (AndrewLinkii) [reply src/main/java/cinema/dao/impl/UserDaoImpl.java:43]: agree(

File: src/main/java/cinema/dao/impl/UserDaoImpl.java [hunk 4]
```diff
@@ -0,0 +1,46 @@
...
+    }
+
+    @Override
+    public Optional<User> findByEmail(String email) {
+        try (Session session = HibernateUtil.getSessionFactory().openSession()) {
+            Query<User> query = session.createQuery("from User where email = :email", User.class);
+            query.setParameter("email", email);
+            return Optional.ofNullable(query.uniqueResult());
+        } catch (Exception e) {
+            throw new DataProcessingException("cant to find user s by login ", e);
```
Comment (dmytro-hryhoruk) [inline src/main/java/cinema/dao/impl/UserDaoImpl.java:43]:
```suggestion
            throw new DataProcessingException("Сan't to find user with email - " + email, e);
```

**Pull request merged**